util/devel/check_paths

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+
+# Copyright 2004-2018 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Try to identify references to non-existent files, e.g. doc/foo
+instead of doc/rst/foo.  This script recognizes potential file
+references by recognizing certain strings as likely to be the prefix
+of a path in the Chapel repository, e.g., "doc/" or "modules/" or
+"test/".
+"""
+
+import os.path
+import argparse
+
+def check_path(p):
+    """returns True if p exists, False if it doesn't"""
+
+    # These are common written-out alternatives, not paths
```

Comment:
We could put these types of exceptions into a file format that is read in, as another future addition.